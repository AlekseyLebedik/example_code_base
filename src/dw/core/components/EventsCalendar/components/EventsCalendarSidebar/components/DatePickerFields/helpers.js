import React from 'react';
import isNil from 'lodash/isNil';

import TextField from '@material-ui/core/TextField';

import {
  changeDateValue,
  renderDatePickerClearDate,
} from 'dw/core/components/EventsCalendar/helpers';

import styles from './index.module.css';

export const endBeforeEndErrorMsg = 'End Date must be after Start Date!';
export const startAfterEndErrorMsg = 'Start Date must be before End Date!';

export const clearOutOfOrderErrorMsg = (
  startBeforeEnd,
  startValueError,
  setStartValueError,
  endValueError,
  setEndValueError
) => {
  if (startBeforeEnd && endValueError === endBeforeEndErrorMsg) {
    setEndValueError(null);
  }
  if (startBeforeEnd && startValueError === startAfterEndErrorMsg) {
    setStartValueError(null);
  }
};

export const datePickerField = (
  label,
  setDate,
  setValue,
  setValueError,
  value,
  valueError
) => (
  <TextField
    autoFocus
    fullWidth
    error={!isNil(valueError)}
    helperText={valueError}
    InputLabelProps={{
      className: styles.eventsCalendarSidebarDatePickerInfoField,
    }}
    InputProps={{
      className: styles.eventsCalendarSidebarDatePickerInfoField,
      endAdornment: renderDatePickerClearDate(value, setDate),
    }}
    label={label}
    onChange={e =>
      changeDateValue(e.target.value, setDate, setValue, setValueError)
    }
    value={value}
  />
);
