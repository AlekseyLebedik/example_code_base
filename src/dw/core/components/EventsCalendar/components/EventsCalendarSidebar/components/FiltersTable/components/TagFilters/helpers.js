import isEmpty from 'lodash/isEmpty';
import uniq from 'lodash/uniq';
import isEqual from 'lodash/isEqual';

// Changes selected status of a custom tag set checkbox
export const toggleTagFilters = (tagFilters, selected) => {
  const newSelected = !selected || selected === 'indeterminate';
  // eslint-disable-next-line no-param-reassign
  tagFilters.unspecified = newSelected;
  if (!isEmpty(tagFilters.userTags)) {
    Object.values(tagFilters.userTags).forEach(userTag => {
      // eslint-disable-next-line no-param-reassign
      userTag.checked = newSelected;
    });
  }
};

// Splits by whitespace, removes commas, and returns tags longer than 1 char
const getValidLengthTags = tags =>
  tags
    .replace(/,/gi, '')
    .split(/[\n\s]+/)
    .filter(tag => tag.length > 1);

// Gets valid tags and returns list of unique tags
export const extractTags = tags => uniq(getValidLengthTags(tags));

// Checks if there are single character tags
const containsSingleCharTags = tags =>
  tags.split(' ').some(tag => tag.length === 1);

// Checks tags list for duplicate tags
const duplicateTagsExist = tags => {
  const tagList = getValidLengthTags(tags).map(tag => tag.toLowerCase());

  return tagList.length !== uniq(tagList).length;
};

// Determine if a tag set already exists
const tagSetExists = (userTags, newTags) =>
  Object.values(userTags).some(tagSet =>
    isEqual(
      tagSet.tags.map(t => t.toLowerCase()).sort(),
      newTags.map(t => t.toLowerCase()).sort()
    )
  );

// Validate a tag set or return appropriate error message if invalid
export const validateTags = (userTags, newTags) => {
  const newTagsNoCommas = newTags.replace(/,/g, '');
  if (containsSingleCharTags(newTagsNoCommas)) {
    return 'All tags must be at least 2 characters!';
  }
  if (duplicateTagsExist(newTagsNoCommas)) {
    return 'You have entered duplicate tags!';
  }
  if (tagSetExists(userTags, extractTags(newTagsNoCommas))) {
    return 'Tag set already exists!';
  }

  return null;
};

// Gets selected status for tag filters, whether true, false, or indeterminate
export const getTagFiltersSelected = tagFilters => {
  if (isEmpty(tagFilters.userTags)) {
    return tagFilters.unspecified;
  }

  const numUserTagsSelected =
    tagFilters.unspecified +
    Object.values(tagFilters.userTags).reduce(
      (acc, userTag) => acc + userTag.checked,
      0
    );

  if (numUserTagsSelected === 0) {
    return false;
  }
  if (numUserTagsSelected === Object.values(tagFilters.userTags).length + 1) {
    return true;
  }
  return 'indeterminate';
};
