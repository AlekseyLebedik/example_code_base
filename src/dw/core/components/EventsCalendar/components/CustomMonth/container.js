/* eslint-disable no-underscore-dangle */
/* eslint-disable camelcase */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment-timezone';
import range from 'lodash/range';
import classnames from 'classnames';
import getHeight from 'dom-helpers/height';
import getPosition from 'dom-helpers/position';
import * as animationFrame from 'dom-helpers/animationFrame';
import { connect } from 'react-redux';

import { formatDateTime } from 'dw/core/helpers/date-time';
import { hasData } from 'dw/core/helpers/object';

import { Navigate } from '@demonware/ct-react-big-calendar';
import DateHeaderGroup from '@demonware/ct-react-big-calendar/lib/DateHeaderGroup';
import * as dates from '@demonware/ct-react-big-calendar/lib/utils/dates';

import DateHeaderGroupWrapper from '../DateHeaderGroupWrapper';

import CustomMonthStateless from './presentational';

export class CustomMonth extends Component {
  static navigate = (date, action, props) => {
    const { eventsCalendarSettings, userTimezone } = props;
    const { numberOfDays } = eventsCalendarSettings;

    switch (action) {
      case Navigate.PREVIOUS:
        return moment(date).tz(userTimezone).subtract(numberOfDays, 'day');

      case Navigate.NEXT:
        return moment(date).tz(userTimezone).add(numberOfDays, 'day');

      default:
        return date;
    }
  };

  static title = date => formatDateTime(date, 'MMMM YYYY');

  constructor(props) {
    super(props);
    this.monthRef = React.createRef();
    this.state = {
      dayViewOverlay: false,
      draggingHeaderGroup: null,
      maxRows: 7,
      needLimitMeasure: true,
      popupEvents: [],
      position: {},
    };
  }

  static getDerivedStateFromProps(props, state) {
    const updates = {};
    if (props.date !== state.date) {
      updates.date = props.date;
    }
    const needLimitMeasure = !dates.eq(state.date, props.date, 'month');
    if (state.needLimitMeasure !== needLimitMeasure) {
      updates.needLimitMeasure = needLimitMeasure;
    }
    return hasData(updates) ? updates : null;
  }

  componentDidMount() {
    if (this.state.needLimitMeasure) this.setRowLimit();

    window.addEventListener(
      'resize',
      (this._resizeListener = () => {
        animationFrame.request(() => {
          this.setState({ needLimitMeasure: true });
        });
      }),
      false
    );

    if (this.props.datePickerProps) {
      document.addEventListener('mouseover', this.onDayHover);
    }
  }

  componentDidUpdate() {
    if (this.state.needLimitMeasure) this.setRowLimit();

    if (this.props.datePickerProps) {
      document.addEventListener('mouseover', this.onDayHover);
    }
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this._resizeListener, false);

    if (this.props.datePickerProps) {
      document.removeEventListener('mouseover', this.onDayHover);
    }
  }

  onDayHover = event => {
    if (
      event.target.className.includes &&
      event.target.className.includes('rbc-day-bg')
    ) {
      const { onHoverDateChange } = this.props.datePickerProps;
      onHoverDateChange(
        event.target[Object.keys(event.target)[0]].return.memoizedProps.value
      );
    }
  };

  getOverlayPlacement = (dayViewOverlay, overlay) => {
    if (dayViewOverlay) {
      if (overlay?.date?.getDay() < 3) return 'right';
      return 'left';
    }
    return 'bottom';
  };

  setPosition = overlay => {
    // Don't show precise time popup window if for repeating event
    if (!overlay?.overlay?.event?.repeat_event_settings)
      this.setState({
        dayViewOverlay: true,
        overlay: {
          ...overlay.overlay,
        },
      });
  };

  setDraggingHeaderGroup = draggingHeaderGroup =>
    this.setState({ draggingHeaderGroup });

  setRowLimit = () => {
    if (this.monthRef && this.monthRef.current) {
      const { clientHeight: eventHeight, parentElement } =
        this.monthRef.current.headingRow;
      const monthRowHeight = getHeight(parentElement.parentElement);
      const newRowLimit = Math.floor(monthRowHeight / eventHeight) - 1;

      this.setState({
        maxRows: newRowLimit,
        needLimitMeasure: false,
      });
    }
  };

  getContainer = () => {
    let element =
      this.monthRef &&
      this.monthRef.current &&
      this.monthRef.current.parentElement;
    if (element) {
      while (element.className !== 'rbc-month-view')
        element = element.parentElement;
      return element;
    }
    return null;
  };

  handleSelectSlot = (slots, slotInfo) => {
    clearTimeout(this.selectTimer);
    this.selectTimer = setTimeout(() =>
      this.props.onSelectSlot({
        action: slotInfo.action,
        start: slots[0],
        end: slots[0],
        slots,
      })
    );
  };

  closePopup = () => {
    this.setState({
      dayViewOverlay: false,
      overlay: null,
    });
  };

  filterWeekEvents = (events, week) =>
    events.filter(
      event =>
        moment(event.start).isSame(week[0], 'week') ||
        moment(event.end).isSame(week[0], 'week') ||
        (moment(event.start).isBefore(week[0], 'day') &&
          moment(event.end).isAfter(week[6], 'day'))
    );

  handleSelectEvent = (...args) => {
    this.clearSelection();
    if (this.props.onSelectEvent) {
      this.props.onSelectEvent.apply(null, [].concat(args));
    }
  };

  handleDateHeaderGroupClick = (group, e, date) => {
    e.preventDefault();
    this.clearSelection();
    this.props.onSelectHeaderGroup(group, e, date);
  };

  monthRange = (customViewOn, numberOfDays, selectedDay) => {
    const start = customViewOn
      ? moment(selectedDay).endOf('day').day(0).subtract(1, 'day')
      : moment(selectedDay)
          .startOf('month')
          .endOf('day')
          .day(0)
          .subtract(1, 'day');
    const end = customViewOn
      ? moment(selectedDay)
          .endOf('day')
          .add(numberOfDays - 1, 'day')
          .endOf('week')
      : moment(selectedDay).endOf('day').endOf('month').endOf('week');
    const numCalendarDays = end.diff(start, 'days');
    return range(0, numCalendarDays).map(() =>
      start.add(1, 'day').startOf('day').toDate()
    );
  };

  handleShowMore = (events, date, cell, slot, target) => {
    clearTimeout(this.selectTimer);
    const { popup } = this.props;
    if (popup) {
      const position = getPosition(cell, this.getContainer());
      this.setState({
        overlay: { date, events, position, target },
        popupEvents: events.filter(
          event =>
            moment(event.start).isSame(moment(date), 'day') ||
            moment(event.end).isSame(moment(date), 'day') ||
            (moment(event.start).isBefore(moment(date), 'day') &&
              moment(event.end).isAfter(moment(date), 'day'))
        ),
        position: cell.getBoundingClientRect(),
      });
    }
  };

  clearSelection() {
    clearTimeout(this.selectTimer);
  }

  renderHeader = ref => {
    const {
      classes,
      components,
      dateHeaderGroups,
      customViewOn,
      numberOfDays,
      selectedDay,
      onDrillDown,
      userTimezone,
    } = this.props;
    const { draggingHeaderGroup } = this.state;
    const { date, key } = ref;
    const day = formatDateTime(date, 'D', userTimezone);
    const isCurrent = moment(date)
      .tz(userTimezone)
      .isSame(moment().tz(userTimezone), 'date');
    const isInSelectedMonth = moment(date)
      .tz(userTimezone)
      .isSame(selectedDay, 'month');
    const isOutsideCustomRange =
      moment(date)
        .tz(userTimezone)
        .endOf('day')
        .isBefore(moment(selectedDay).startOf('day')) ||
      moment(date)
        .tz(userTimezone)
        .startOf('day')
        .isAfter(
          moment(selectedDay)
            .add(numberOfDays - 1, 'day')
            .endOf('day')
        );
    const isOffRange =
      (customViewOn && isOutsideCustomRange) ||
      (!customViewOn && !isInSelectedMonth);
    const isSelectedDay = moment(date)
      .tz(userTimezone)
      .isSame(moment(selectedDay).tz(userTimezone), 'date');
    const label =
      day === '1' && customViewOn
        ? `${formatDateTime(date, 'MMMM D', userTimezone)}`
        : day;

    const headerClasses = classnames(
      'rbc-date-cell',
      classes.dayHeader,
      isOffRange && 'rbc-off-range',
      isCurrent && classes.dayHeaderCurrent,
      isSelectedDay && !isCurrent && classes.dayHeaderSelected
    );

    const DateHeaderGroupComponent =
      components.dateHeaderGroup || DateHeaderGroup;
    const GroupWrapper =
      components.dateHeaderGroupWrapper || DateHeaderGroupWrapper;

    const headerGroups = [...dateHeaderGroups];
    if (draggingHeaderGroup) headerGroups.push(draggingHeaderGroup);
    const groupsOnDate = headerGroups.filter(g =>
      moment(g.start)
        .tz(userTimezone)
        .isSame(moment(date).tz(userTimezone), 'date')
    );

    const Groups = (
      <div className="rbc-date-header-groups">
        {groupsOnDate.map(group => (
          <GroupWrapper
            group={group}
            key={`${group.id || group.name}${group.__isPreview && '/preview'}`}
          >
            <DateHeaderGroupComponent
              components={components}
              group={group}
              onSelect={(g, e) => this.handleDateHeaderGroupClick(g, e, date)}
            />
          </GroupWrapper>
        ))}
      </div>
    );

    return (
      <div key={key} className={headerClasses}>
        <a
          onClick={e => {
            clearTimeout(this.selectTimer);
            onDrillDown(e);
          }}
          value={date}
        >
          {label}
        </a>
        {Groups}
      </div>
    );
  };

  render() {
    const newProps = {
      ...this.props,
      ...this.state,
      closePopup: this.closePopup,
      filterWeekEvents: this.filterWeekEvents,
      getContainer: this.getContainer,
      getOverlayPlacement: this.getOverlayPlacement,
      handleSelectSlot: this.handleSelectSlot,
      handleSelectEvent: this.handleSelectEvent,
      handleShowMore: this.handleShowMore,
      monthRange: this.monthRange,
      monthRef: this.monthRef,
      navigate: this.navigate,
      renderHeader: this.renderHeader,
      setDraggingHeaderGroup: this.setDraggingHeaderGroup,
      setRef: this.setRef,
      setPosition: this.setPosition,
      title: this.title,
      customViewOn: this.props.customViewOn,
      numberOfDays: this.props.numberOfDays,
      selectedDay: this.props.selectedDay,
    };
    return <CustomMonthStateless {...newProps} />;
  }
}

CustomMonth.propTypes = {
  classes: PropTypes.object,
  components: PropTypes.object.isRequired,
  dateHeaderGroups: PropTypes.arrayOf(PropTypes.object),
  date: PropTypes.instanceOf(Date).isRequired,
  events: PropTypes.arrayOf(PropTypes.object).isRequired,
  customViewOn: PropTypes.bool.isRequired,
  datePickerProps: PropTypes.object,
  numberOfDays: PropTypes.number.isRequired,
  selectedDay: PropTypes.object.isRequired,
  localizer: PropTypes.object.isRequired,
  onDrillDown: PropTypes.func.isRequired,
  onSelectEvent: PropTypes.func.isRequired,
  onSelectHeaderGroup: PropTypes.func,
  onSelectSlot: PropTypes.func.isRequired,
  popup: PropTypes.bool.isRequired,
  userTimezone: PropTypes.string.isRequired,
};

CustomMonth.defaultProps = {
  classes: {},
  dateHeaderGroups: [],
  onSelectHeaderGroup: () => {},
  datePickerProps: {},
};
export default connect(state => ({
  customViewOn: state.Core.EventsCalendar.customViewOn,
  numberOfDays: state.Core.EventsCalendar.numberOfDays,
  selectedDay: state.Core.EventsCalendar.selectedDay,
  datePickerProps: state.Core.EventsCalendar.datePickerProps,
}))(CustomMonth);
