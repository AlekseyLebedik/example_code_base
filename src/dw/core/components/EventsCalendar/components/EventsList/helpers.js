import moment from 'moment';
import { formatDateTime } from 'dw/core/helpers/date-time';
import { dateComparator } from 'dw/core/helpers/aggrid';

export const customDateComparator = (date1, date2) => {
  if (!date1) return 1;
  if (!date2) return -1;
  return dateComparator(date1, date2);
};

export const handleEndDate = (start, end) => !moment(start).isSame(end) && end;

export const dateValueGetter = date => (date ? formatDateTime(date) : '---');

export const eventWriteCheck = (event, writeAccess) =>
  writeAccess &&
  event.status === 'open' &&
  !event.isReadOnly &&
  !event.is_manually_locked &&
  !event.is_restricted;
