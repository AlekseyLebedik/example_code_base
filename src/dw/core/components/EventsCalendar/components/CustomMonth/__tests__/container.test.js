import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment-timezone';
import range from 'lodash/range';

import { dateToUTCTimestamp } from 'dw/core/helpers/date-time';
import { customMonthTestProps as props } from 'dw/core/components/EventsCalendar/testData';

import { CustomMonth } from '../container';

describe('CustomMonth', () => {
  let instance;
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CustomMonth {...props} />);
    instance = wrapper.instance();
  });

  it('renders default', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('filterWeekEvents filters out events not in week', () => {
    const week = [
      moment('2018-12-30T00:00:00Z').toDate(),
      moment('2018-12-31T00:00:00Z').toDate(),
      moment('2019-01-01T00:00:00Z').toDate(),
      moment('2019-01-02T00:00:00Z').toDate(),
      moment('2019-01-03T00:00:00Z').toDate(),
      moment('2019-01-04T00:00:00Z').toDate(),
      moment('2019-01-05T00:00:00Z').toDate(),
    ];

    expect(instance.filterWeekEvents(props.events, week)).toEqual([
      {
        created_at: dateToUTCTimestamp(new Date(2019, 0, 1)),
        end: moment('2019-01-01T00:00:00Z').toDate(),
        publish_at: dateToUTCTimestamp(new Date(2019, 0, 1)),
        start: moment('2019-01-01T00:00:00Z').toDate(),
        status: 'pending',
        title: 'Test 1',
        type: 'eventManager',
      },
      {
        created_at: dateToUTCTimestamp(new Date(2019, 0, 1)),
        end: moment('2019-01-01T00:00:00Z').toDate(),
        publish_at: dateToUTCTimestamp(new Date(2019, 0, 1)),
        start: moment('2019-01-01T00:00:00Z').toDate(),
        status: 'approved',
        title: 'Test 2',
        type: 'eventManager',
      },
      {
        created_at: dateToUTCTimestamp(new Date(2019, 0, 1)),
        end: moment('2019-01-01T00:00:00Z').toDate(),
        publish_at: dateToUTCTimestamp(new Date(2019, 0, 1)),
        start: moment('2019-01-01T00:00:00Z').toDate(),
        status: 'pending',
        title: 'Test 3',
        type: 'eventManager',
      },
    ]);
  });

  describe('monthRange', () => {
    it('monthRange returns the correct date range for a standard month view', () => {
      const { customViewOn, numberOfDays, selectedDay } =
        props.eventsCalendarSettings;
      let currentDate = moment('2018-12-29T00:00:00Z').toDate();
      const expectedRange = range(0, 35).map(() => {
        currentDate = moment(currentDate).startOf('day').add(1, 'day').toDate();
        return currentDate;
      });

      expect(
        instance.monthRange(customViewOn, numberOfDays, selectedDay)
      ).toEqual(expectedRange);
    });

    it('monthRange returns the correct date range for a custom month view', () => {
      const updatedCalendarSettings = {
        customViewOn: true,
        numberOfDays: 20,
        selectedDay: moment(new Date(2019, 0, 1)).tz('UTC'),
      };
      wrapper.setProps({
        eventsCalendarSettings: updatedCalendarSettings,
      });
      let currentDate = moment('2018-12-29T00:00:00Z').toDate();
      const expectedRange = range(0, 28).map(() => {
        currentDate = moment(currentDate).startOf('day').add(1, 'day').toDate();
        return currentDate;
      });

      expect(
        instance.monthRange(
          updatedCalendarSettings.customViewOn,
          updatedCalendarSettings.numberOfDays,
          updatedCalendarSettings.selectedDay
        )
      ).toEqual(expectedRange);
    });
  });
});
