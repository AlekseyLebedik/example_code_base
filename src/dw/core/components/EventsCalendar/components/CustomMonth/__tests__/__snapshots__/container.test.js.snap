// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomMonth renders default 1`] = `
<CustomMonthStateless
  accessors={Object {}}
  addCustomTagSet={[MockFunction]}
  changeCalendarView={[MockFunction]}
  changeNumberOfDays={[MockFunction]}
  classes={Object {}}
  clearEventsCalendarSettings={[MockFunction]}
  closePopup={[Function]}
  components={Object {}}
  copyLinkToClipboard={[MockFunction]}
  createEventModalId="CREATE_EVENT"
  currentProject={
    Object {
      "id": 1,
    }
  }
  customFiltersSelected={false}
  customRangeError=""
  customRangeInput={30}
  customViewOn={false}
  date={2019-01-01T00:00:00.000Z}
  dateHeaderGroups={Array []}
  datePickerProps={Object {}}
  dayViewOverlay={false}
  daysWithDots={Object {}}
  defaultDate={"2019-01-01T00:00:00.000Z"}
  disabledFilters={Object {}}
  displayView="calendar"
  draggingHeaderGroup={null}
  envTypeFilters={
    Object {
      "Certification": true,
      "Development": true,
      "Live": true,
    }
  }
  eventGroups={
    Array [
      Object {
        "EventDetails": [Function],
        "GroupLink": [Function],
        "classes": Object {
          "applied": "applied-style",
          "approved": "approved-style",
          "cancelled": "cancelled-style",
          "expired": "expired-style",
          "pending": "pending-style",
          "rejected": "rejected-style",
          "scheduled": "scheduled-style",
          "test-failed": "test-failed-style",
          "testing": "testing-style",
        },
        "customEventStyles": [Function],
        "eventDragDrop": [MockFunction],
        "eventTypes": Object {
          "eventManager": Object {
            "children": Object {
              "checkedByMe": Object {
                "children": null,
                "name": "Checked By Me",
                "selectedByDefault": false,
              },
              "status": Object {
                "children": Object {
                  "applied": Object {
                    "children": null,
                    "name": "Applied",
                    "selectedByDefault": true,
                  },
                  "approved": Object {
                    "children": null,
                    "name": "Approved",
                    "selectedByDefault": true,
                  },
                  "cancelled": Object {
                    "children": null,
                    "name": "Cancelled",
                    "selectedByDefault": true,
                  },
                  "expired": Object {
                    "children": null,
                    "name": "Expired",
                    "selectedByDefault": true,
                  },
                  "pending": Object {
                    "children": null,
                    "name": "Pending",
                    "selectedByDefault": true,
                  },
                  "rejected": Object {
                    "children": null,
                    "name": "Rejected",
                    "selectedByDefault": true,
                  },
                  "scheduled": Object {
                    "children": null,
                    "name": "Scheduled",
                    "selectedByDefault": true,
                  },
                  "test-failed": Object {
                    "children": null,
                    "name": "Test Failed",
                    "selectedByDefault": true,
                  },
                  "testing": Object {
                    "children": null,
                    "name": "Testing",
                    "selectedByDefault": true,
                  },
                },
                "name": "Status",
              },
            },
            "name": "Event Manager",
            "selectedByDefault": true,
          },
        },
        "events": Array [
          Object {
            "allDay": false,
            "end": 1555460772,
            "id": 1,
            "publish_at": 1555460772,
            "start": 1555460772,
            "status": "approved",
            "title": "test",
            "type": "eventManager",
          },
          Object {
            "allDay": false,
            "end": 1555460772,
            "id": 1,
            "publish_at": 1555460772,
            "start": 1555460772,
            "status": "pending",
            "title": "test 2",
            "type": "eventManager",
          },
        ],
        "loading": Object {
          "error": null,
          "isLoading": false,
        },
        "modifyEvent": [MockFunction],
        "type": "eventManager",
        "wrapper": [MockFunction],
      },
    ]
  }
  eventTimeOffset={0}
  events={
    Array [
      Object {
        "created_at": 1546300800,
        "end": 2019-01-01T00:00:00.000Z,
        "publish_at": 1546300800,
        "start": 2019-01-01T00:00:00.000Z,
        "status": "pending",
        "title": "Test 1",
        "type": "eventManager",
      },
      Object {
        "created_at": 1546300800,
        "end": 2019-01-01T00:00:00.000Z,
        "publish_at": 1546300800,
        "start": 2019-01-01T00:00:00.000Z,
        "status": "approved",
        "title": "Test 2",
        "type": "eventManager",
      },
      Object {
        "created_at": 1546300800,
        "end": 2019-01-01T00:00:00.000Z,
        "publish_at": 1546300800,
        "start": 2019-01-01T00:00:00.000Z,
        "status": "pending",
        "title": "Test 3",
        "type": "eventManager",
      },
      Object {
        "created_at": 1546905600,
        "end": 2019-01-08T00:00:00.000Z,
        "publish_at": 1546905600,
        "start": 2019-01-08T00:00:00.000Z,
        "status": "pending",
        "title": "Test 4",
        "type": "eventManager",
      },
    ]
  }
  eventsCalendarSettings={
    Object {
      "customViewOn": false,
      "displayView": "calendar",
      "envTypeFilters": Object {
        "Certification": true,
        "Development": true,
        "Live": true,
      },
      "eventTimeOffset": 0,
      "filters": Object {
        "customTags": Object {
          "unspecified": true,
          "userTags": Object {
            "0": Object {
              "checked": true,
              "tags": Array [
                "PS4",
                "XBOX",
              ],
            },
            "1": Object {
              "checked": true,
              "tags": Array [
                "COD",
                "BO4",
              ],
            },
          },
        },
        "environments": Object {
          "Certification": true,
          "Cross Environment": true,
          "Development": true,
          "Live": true,
        },
        "gamertags": Object {},
        "platforms": Object {
          "CROSSPLAY": true,
          "PC": true,
          "PS4": true,
          "SWITCH": true,
          "Unspecified": true,
          "XB1": true,
        },
        "projects": Object {
          "Call of Duty: Modern Warfare 2": true,
          "Cross Project": true,
          "GTR Project": true,
          "Permissions: Test 1": true,
        },
        "sources": Object {
          "abTesting": Object {
            "active": false,
            "analysis": false,
            "archived": false,
            "config": false,
            "killed": false,
          },
          "demonwareEvents": Object {
            "criticalEvents": true,
            "generalComments": true,
            "incidents": true,
            "maintenance": true,
          },
          "eventManager": Object {
            "applied": false,
            "approved": true,
            "cancelled": false,
            "expired": false,
            "pending": false,
            "rejected": false,
            "scheduled": false,
            "test-failed": false,
            "testing": false,
          },
          "externalEvents": Object {
            "holidays": true,
            "pmg": Object {
              "Other": true,
            },
            "sports": false,
            "video-games": false,
          },
          "informationalEvents": Object {
            "demonware": true,
          },
        },
        "stories": undefined,
      },
      "groupLoadingStatuses": Object {
        "abTesting": Object {
          "error": null,
          "isLoading": false,
        },
        "demonwareEvents": Object {
          "error": null,
          "isLoading": false,
        },
        "eventManager": Object {
          "error": null,
          "isLoading": false,
        },
        "externalEvents": Object {
          "error": null,
          "isLoading": false,
        },
        "informationalEvents": Object {
          "error": null,
          "isLoading": false,
        },
      },
      "numberOfDays": 30,
      "selectedDay": "2019-01-01T00:00:00.000Z",
      "selectedStyle": "sources",
      "selectedView": "month",
    }
  }
  filterWeekEvents={[Function]}
  filteredEvents={
    Array [
      Object {
        "allDay": false,
        "end": 1555460772,
        "id": 1,
        "publish_at": 1555460772,
        "start": 1555460772,
        "status": "approved",
        "title": "test",
        "type": "eventManager",
      },
    ]
  }
  filteredGamertagGroups={Array []}
  filters={
    Object {
      "customTags": Object {
        "unspecified": true,
        "userTags": Object {
          "0": Object {
            "checked": true,
            "tags": Array [
              "PS4",
              "XBOX",
            ],
          },
          "1": Object {
            "checked": true,
            "tags": Array [
              "COD",
              "BO4",
            ],
          },
        },
      },
      "environments": Object {
        "Certification": true,
        "Cross Environment": true,
        "Development": true,
        "Live": true,
      },
      "gamertags": Object {},
      "platforms": Object {
        "CROSSPLAY": true,
        "PC": true,
        "PS4": true,
        "SWITCH": true,
        "Unspecified": true,
        "XB1": true,
      },
      "projects": Object {
        "Call of Duty: Modern Warfare 2": true,
        "Cross Project": true,
        "GTR Project": true,
        "Permissions: Test 1": true,
      },
      "sources": Object {
        "abTesting": Object {
          "active": false,
          "analysis": false,
          "archived": false,
          "config": false,
          "killed": false,
        },
        "demonwareEvents": Object {
          "criticalEvents": true,
          "generalComments": true,
          "incidents": true,
          "maintenance": true,
        },
        "eventManager": Object {
          "applied": false,
          "approved": true,
          "cancelled": false,
          "expired": false,
          "pending": false,
          "rejected": false,
          "scheduled": false,
          "test-failed": false,
          "testing": false,
        },
        "externalEvents": Object {
          "holidays": true,
          "pmg": Object {
            "Other": true,
          },
          "sports": false,
          "video-games": false,
        },
        "informationalEvents": Object {
          "demonware": true,
        },
      },
      "stories": undefined,
    }
  }
  getContainer={[Function]}
  getDrilldownView={[MockFunction]}
  getNow={[MockFunction]}
  getOverlayPlacement={[Function]}
  getters={Object {}}
  groupLoadingStatuses={
    Object {
      "abTesting": Object {
        "error": null,
        "isLoading": false,
      },
      "demonwareEvents": Object {
        "error": null,
        "isLoading": false,
      },
      "eventManager": Object {
        "error": null,
        "isLoading": false,
      },
      "externalEvents": Object {
        "error": null,
        "isLoading": false,
      },
      "informationalEvents": Object {
        "error": null,
        "isLoading": false,
      },
    }
  }
  handleOpenCreateEventModal={[MockFunction]}
  handleSelectEvent={[Function]}
  handleSelectSlot={[Function]}
  handleShowMore={[Function]}
  history={
    Object {
      "replace": [MockFunction],
    }
  }
  localizer={
    Object {
      "format": [MockFunction],
    }
  }
  location={
    Object {
      "search": "",
    }
  }
  maxRows={7}
  monthRange={[Function]}
  monthRef={
    Object {
      "current": null,
    }
  }
  navigateCalendar={[MockFunction]}
  needLimitMeasure={true}
  numberOfDays={30}
  onChangeDatePickerSelection={[MockFunction]}
  onDrillDown={[MockFunction]}
  onEventDrop={[MockFunction]}
  onFetchEvents={[MockFunction]}
  onNavigate={[MockFunction]}
  onPresetsUpdate={[MockFunction]}
  onSelectEvent={[MockFunction]}
  onSelectHeaderGroup={[Function]}
  onSelectSlot={[MockFunction]}
  onSetCalendarSettings={[MockFunction]}
  permissions={
    Object {
      "adminPermission": true,
      "adminUser": true,
      "emUser": true,
      "eventWritePermission": true,
      "staffUser": true,
    }
  }
  platforms={
    Array [
      "CROSSPLAY",
      "PC",
      "PS4",
      "SWITCH",
      "XB1",
    ]
  }
  popup={true}
  popupEvents={Array []}
  position={Object {}}
  presetOptions={Array []}
  projectId="1"
  renderHeader={[Function]}
  selectedDay={"2019-01-01T00:00:00.000Z"}
  selectedStyle="sources"
  selectedView="month"
  setCalendarSettings={[MockFunction]}
  setCustomEvent={[MockFunction]}
  setDayMonthStyle={[MockFunction]}
  setDraggingHeaderGroup={[Function]}
  setEventStyle={[MockFunction]}
  setNonCustomDays={[MockFunction]}
  setPosition={[Function]}
  showAllColors={false}
  sidebar={false}
  toastMsg={[MockFunction]}
  toggleDisplayView={[MockFunction]}
  userTimezone="UTC"
  views={
    Object {
      "day": Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(CustomWeek)",
        "navigate": [Function],
        "title": [Function],
        "type": [Function],
      },
      "month": Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(CustomMonth)",
        "navigate": [Function],
        "title": [Function],
        "type": [Function],
      },
      "week": Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(CustomWeek)",
        "navigate": [Function],
        "title": [Function],
        "type": [Function],
      },
    }
  }
/>
`;
